myuw/dao/student.py

```diff
@@ -0,0 +1,132 @@
+"""
+This module will provide a method to retrieve all user majors, class standings
+and minors along with current and rollup variables to facilitate
+determinations about user context in the case that a user is not
+enrolled for the current quarter
+"""
+from myuw.dao.enrollment import enrollment_search
+from uw_sws.term import get_current_term
+
+
+def get_student_status(request):
+    """
+    This will return a dictionary with the student status over time alongside
+    various rollup variables
+    :param request:
+    :return: dict
+    """
+    enrollments = enrollment_search(request)
+
+    return {
```

Comment:
Sorry for being so high maintenance. 
Could you refactor the code to go through the enrollments once, and gather the major, minor, at the same time?

Reply:
Yeah, that's a good idea! Thanks for the tip.